river/stats/kolmogorov_smirnov.py

```diff
@@ -0,0 +1,288 @@
+    material of the authors of paper [^1], at
+    [the following Github repository](https://github.com/denismr/incremental-ks/tree/master).
+
+    Parameters
+    ----------
+    statistic
+        The method used to calculate the statistic, can be either "ks" or "kuiper".
+        The default value is set as "ks".
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> from river import stats
+
+    >>> stream_a = [1, 1, 2, 2, 3, 3, 4, 4]
+    >>> stream_b = [1, 1, 1, 1, 2, 2, 2, 2]
+
+    >>> incremental_ks = stats.KolmogorovSmirnov(statistic="ks")
+    >>> for a, b in zip(stream_a, stream_b):
+    ...     incremental_ks.update(a, b)
+
+    >>> incremental_ks
+    KolmogorovSmirnov: 0.5
+
+    >>> incremental_ks.n_samples
+    8
+
+    References
+    ----------
+    [^1]: dos Reis, D.M. et al. (2016) ‘Fast unsupervised online drift detection using incremental Kolmogorov-Smirnov
+    test’, Proceedings of the 22nd ACM SIGKDD International Conference on Knowledge Discovery and Data Mining.
+    doi:10.1145/2939672.2939836.
+    [^2]: C. R. Aragon and R. G. Seidel. Randomized search trees. In FOCS, pages 540–545. IEEE, 1989.
+    [^3]: Kuiper, N. H. (1960). "Tests concerning random points on a circle".
+    Proceedings of the Koninklijke Nederlandse Akademie van Wetenschappen, Series A. 63: 38–47.
+    """
+
+    def __init__(self, statistic="ks"):
+        self.treap = None
+        self.n_samples = 0
+        self.statistic = statistic
+
+    @staticmethod
+    def ca(p_value):
+        return (-0.5 * math.log(p_value)) ** 0.5
+
+    @classmethod
+    def ks_threshold(cls, p_value, n_samples):
+        return cls.ca(p_value) * (2.0 * n_samples / n_samples**2)
+
+    def update(self, x, y):
+        keys = ((x, 0), (y, 1))
+
+        self.n_samples += 1
+
+        for key in keys:
+            left, left_g, right, val = None, None, None, None
+
+            left, right = Treap.split_keep_right(self.treap, key)
+            left, left_g = Treap.split_greatest(left)
+            val = 0 if left_g is None else left_g.value
+
+            left = Treap.merge(left, left_g)
+            right = Treap.merge(Treap(key, val), right)
+            Treap.sum_all(right, 1 if key[1] == 0 else -1)
+
+            self.treap = Treap.merge(left, right)
+
+    def revert(self, x, y):
+        keys = ((x, 0), (y, 1))
+
+        self.n_samples -= 1
+
+        for key in keys:
+            left, right, right_l = None, None, None
+
+            left, right = Treap.split_keep_right(self.treap, key)
+            right_l, right = Treap.split_smallest(right)
+
+            if right_l is not None and right_l.key == key:
+                Treap.sum_all(right, -1 if key[1] == 0 else 1)
+            else:
+                right = Treap.merge(right_l, right)
+
+            self.treap = Treap.merge(left, right)
+
+    def get(self):
+        assert self.statistic in ["ks", "kuiper"]
```

Comment:
You can remove the assert here. What you should do is add an `else` to handle an unknown string

Reply:
Totally agree! Suggestion committed.